packages/kbn-optimizer/src/integration_tests/basic_optimization.test.ts

```diff
@@ -101,12 +101,12 @@ it('builds expected bundles, saves bundle counts to metadata', async () => {
 
   const runningStates = msgs.filter(msg => msg.state.phase === 'running');
   assert(
-    'produce two or three "running" states',
-    runningStates.length === 2 || runningStates.length === 3
+    'produce four or five "running" states',
```

Comment:
Would it be useful to add a comment about the reason for the different number of messages?

Reply:
TBH, I don't really like or know what these tests are useful for. @spalger could you provide any insight into why we have to have this OR logic in the assertions? Is this flaky? Is there something we should be waiting to complete to make this more deterministic?

Reply:
Yeah, the number of possible running states is based on how quickly the builds take which makes them non-deterministic. I'm fine dropping this assertion personally.